pkg/clusteragent/admission/mutate/agent_sidecar/agent_sidecar.go

```diff
@@ -155,6 +171,14 @@ func (w *Webhook) injectAgentSidecar(pod *corev1.Pod, _ string, _ dynamic.Interf
 	if !agentSidecarExists {
 		agentSidecarContainer := w.getDefaultSidecarTemplate()
 		pod.Spec.Containers = append(pod.Spec.Containers, *agentSidecarContainer)
+		if w.isReadOnlyRootFilesystem() {
+			w.addDefaultSidecarSecurity(agentSidecarContainer)
+			pod.Spec.Volumes = append(pod.Spec.Volumes, *w.getDefaultSidecarVolumeTemplate())
+			// Don't want to apply any overrides to the agent sidecar init container
+			defer func() {
```

Comment:
Question: why do we need a defer function here?

Reply:
If we add the initContainer to the structure here. Then the `applyProviderOverrides` will affect the init container's definition. This specifically happens at `applyProviderOverrides` in the `fargate` case when `applyFargateOverrides` calls `common.InjectVolume(pod, volume, volumeMount)` and it adds a VolumeMount to that initContainers [here](https://github.com/DataDog/datadog-agent/blob/48a09b8d04a9b2313137a7ce4f29053038856575/pkg/clusteragent/admission/mutate/common/common.go#L141). This initContainer does not need this VolumeMount for APM and DogStatsD sockets.